install.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+# Copyright 2018 the Deno authors. All rights reserved. MIT license.
+# TODO(everyone): Keep this script simple and easily auditable.
+
+set -e
+
+if [ $# -eq 0 ]; then
+  version="v0.2.6"
+else
+  version=$1
+fi
+
+deno_dir="$HOME/.deno/bin"
+
+if [ ! -d "$deno_dir" ]; then
+  mkdir -p "$deno_dir"
+fi
+
+case $(uname -s) in
+  Darwin) os="osx" ;;
+  *) os="linux" ;;
+esac
+
+deno_uri="https://github.com/denoland/deno/releases/download/${version}/deno_${os}_x64.gz"
+
+curl -fL# -o "$deno_dir/deno.gz" "$deno_uri"
+gunzip -df "$deno_dir/deno.gz"
+chmod +x "$deno_dir/deno"
+
+echo "Deno was installed successfully."
+case $PATH in
+  *"$deno_dir"*) echo "Run 'deno --help' to get started." ;;
+  *) echo "Run '~/.deno/bin/deno --help' to get started." ;;
+esac
```

Comment:
For extra safety wrap those paths with `:`:
```sh
case ":$PATH:" in
  *:$deno_dir:*) echo "Run 'deno --help' to get started." ;;
  *) echo "Run '~/.deno/bin/deno --help' to get started." ;;
esac
```

Reply:
Wrapping in `:` won't help if it's the last or first path.

Actually I'd need to check `$HOME/.deno/bin`, `$HOME/.deno/bin/`, `$HOME/.deno/bin:*`, `$HOME/.deno/bin/:*`, `*:$HOME/.deno/bin:*`, `*:$HOME/.deno/bin/:*`, `*:$HOME/.deno/bin`, `*:$HOME/.deno/bin/`, and all of those variants with `$HOME` being replaced by `~`.

That would be perfectly correct, but hard to maintain. I'd prefer to just check `*$HOME/.deno/bin*` and `*~/.deno/bin*` for the sake of maintainability.

Reply:
> Wrapping in `:` won't help if it's the last or first path.

I think you missed that `":$PATH:"` is wrapped too, not just `*:$deno_dir:*`?

Reply:
In the `pwsh` script, I split the PATH by `:` and then check the splitted paths for `$HOME/.deno/bin`, `$HOME/.deno/bin/`, `~/.deno/bin` and `~/.deno/bin/`. I just don't know how to do this nicely in `sh`.

Reply:
Checking for `~/.deno/bin` seems awkward 🙃
Only way for unexpanded `~` to get inside path is doing something like this:
```sh
PATH="~/.deno/bin":$PATH
```
by specifically instructing shell not to expand path by quoting it. I firmly believe this may end up in your path only by mistake and we shouldn't account for _wrongly_ declared `PATH`s?

Reply:
> In the `pwsh` script, I split the PATH by `:` and then check the splitted paths for `$HOME/.deno/bin`, `$HOME/.deno/bin/`, `~/.deno/bin` and `~/.deno/bin/`. I just don't know how to do this nicely in `sh`.

You are looking it from different perspective. I'm not talking about splitting and checking if resulting array contains matches. What I did in my snippet is:
1. wrapping original `PATH` value with surrounding `:`
2. finding `:<target>:` substring in result of 1.

Reason why would you like to wrap is because without it you are matching against some portion of a path or a full one:
```sh
*$deno_dir*) ;;
# would match: `/usr/bin:$deno_dir:/usr/local/bin`
# but also: `/usr/bin:$deno_dir/some/trailing/stuff:/usr/local/bin`
# and this ofc: `/usr/bin:/parent/dirs/$deno_dir:/usr/local/bin`
```
while:
```sh
*:$deno_dir:*) ;;
# would match: `/usr/bin:$deno_dir:/usr/local/bin`
# would NOT match: `/usr/bin:$deno_dir/some/trailing/stuff:/usr/local/bin`
```

Reply:
> I firmly believe this may end up in your path only by mistake

I see. I'll fix that.

> wrapping original PATH value with surrounding :

I missed that. Thanks!

Reply:
@vladimyr Am I doing the wrong check maybe? Would it perhaps be better to check whether deno is available on the PATH by running `which deno`?

Reply:
I used `type` instead of `which` now. What do you think, @vladimyr?

Reply:
From my experience, reading other projects installation scripts, people typically use `command -v <command>` but I don't think using `type` instead is bad either.

Reply:
Thank you, @vladimyr! Fixed it.